tests/auto/instantiatetypes/tst_instantiatetypes.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+  Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Nathan Collins, nathan.collins@kdab.com
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <QtTest>
+
+#include <QDirIterator>
+#include <QtQml>
+
+class tst_InstantiateTypes : public QObject
+{
+    Q_OBJECT
+
+public:
+    tst_InstantiateTypes();
+
+private slots:
+    void creatableTypes();
+    void uncreatableTypes();
+
+private:
+    QQmlEngine* m_qmlEngine;
+
+};
+
+tst_InstantiateTypes::tst_InstantiateTypes()
+    : QObject()
+    , m_qmlEngine(new QQmlEngine(this))
+{
+    // Add extensionplugin import path
+    m_qmlEngine->addImportPath(QStringLiteral("%1/../../../qml").arg(QCoreApplication::applicationDirPath()));
+}
+
+void tst_InstantiateTypes::creatableTypes()
+{
+    QDirIterator iterator(QStringLiteral(":/qml/creatable/"), QDirIterator::Subdirectories);
+    while (iterator.hasNext())
+    {
```

Comment:
lets go with the Qt coding style for newly added code: https://wiki.qt.io/Qt_Coding_Style
i.e. in this case opening brace on the line of the while condition